ghastly/read_input.py

```diff
@@ -116,6 +166,35 @@ def create_sim_block(self, core_intake, core_main, core_outtake):
             Ghastly sim object containing simulation parameters and core
             objects.
         '''
+
+        hz_case = self.sim_var.get("recirc_hz")
+        match hz_case:
+            case None:
+                recirc_hz = 1
+            case _:
+                recirc_hz = self.sim_var["recirc_hz"]
+        
+        target_case = self.sim_var.get("recirc_target")
+        if type(target_case) != int and target_case != None:
+            raise TypeError('''The target number of pebbles to recirculate
+                            should be an integer''')
+        match target_case:
+            case None:
+                recirc_target = 1
+            case _:
+                recirc_target = self.sim_var["recirc_target"]
+
+        fidelity_case = self.sim_var.get("fidelity")
```

Comment:
Why doesn't sim have an attribute called fidelity? Getting variables with string keys is a little fragile compared to just setting up attributes on an object.